test/auth.e2e-spec.ts

```diff
@@ -0,0 +1,86 @@
+import * as request from 'supertest';
+import { Test } from '@nestjs/testing';
+import { AuthModule } from '../src/auth/auth.module';
+import { AuthService } from '../src/auth/auth.service';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { UserService } from '../src/user/user.service';
+import { PrismaService } from '../src/prisma/prisma.service';
+import { Role } from '@prisma/client';
+
+describe('AuthController (e2e)', () => {
+  let app: INestApplication;
+
+  const mockUserRequest = {
+    email: 'test@gmail.com',
+    password: 'testpassword123',
+  };
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [AuthModule],
+      providers: [AuthService, UserService, PrismaService],
+    }).compile();
+
+    app = moduleRef.createNestApplication();
+    await app.init();
+  });
+
+  describe('/auth/signup (POST)', () => {
+    it('should register a user and return the new user object', async () => {
+      request(app.getHttpServer())
+        .post('/auth/signup')
+        .set('Accept', 'application/json')
+        .send(mockUserRequest)
+        .expect(HttpStatus.FORBIDDEN);
+    });
```

Comment:
something is not right in this test - you say it `should register a user and return the new user object` but it excepts 403 Forbidden status and not checking response body if the user object exists.